### PR TITLE
[web] Allow some more nulls in platform_dispatcher.

### PR DESCRIPTION
**Pull request by ditman**
* Allow nulls in setApplicationSwitcherDescription
* Tighten setPreferredOrientation types.

Fixes https://github.com/flutter/flutter/issues/83351

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart [hunk 1]
```diff
@@ -396,9 +396,11 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
                 callback, codec.encodeSuccessEnvelope(true));
             return;
           case 'SystemChrome.setApplicationSwitcherDescription':
-            final Map<String, dynamic> arguments = decoded.arguments;
-            domRenderer.setTitle(arguments['label']);
-            domRenderer.setThemeColor(ui.Color(arguments['primaryColor']));
+            final Map<String, Object> arguments = decoded.arguments;
+            final String label = arguments['label'] as String? ?? '';
+            final int primaryColor = arguments['primaryColor'] as int? ?? 0;
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/platform_dispatcher.dart:401]: Are there any breadcrumbs we could leave so we know where these defaults are coming from? For example, in other places we link to the C++ variant engine code and advise that people keep the values in sync.

Comment (ditman) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:401]: Let me check what the mobile engine does, and I'll reproduce those values there. I'll comment to the relevant implementation detail too.

Comment (ditman) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:401]: [iOS does nothing](https://github.com/flutter/engine/blob/138c91c614d742c52aa5432b4cb921f0ff9fdee2/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm#L155-L157), and android [passes values through](https://github.com/flutter/engine/blob/138c91c614d742c52aa5432b4cb921f0ff9fdee2/shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java#L185-L205).

The only documentation I found about this, in the Android docs, is that the primaryColor *must be opaque*, [here](https://developer.android.com/reference/kotlin/android/app/ActivityManager.TaskDescription#TaskDescription(kotlin.String,%20kotlin.Int,%20kotlin.Int)). This is used in Android to customize the name and color of the app [when task-switching](https://stackoverflow.com/questions/44682593/how-do-i-style-the-window-bar-when-my-android-app-is-in-background).

I'll make this an opaque black when null, as you suggested: `0xFF000000`

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart [hunk 2]
```diff
@@ -396,9 +396,11 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
...
+            domRenderer.setTitle(label);
+            domRenderer.setThemeColor(ui.Color(primaryColor));
```
[the remark this answers is not in the document]
Comment (ditman) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:403]: (Merging this thread with the one above, about behavior of the mobile engine)

File: lib/web_ui/lib/src/engine/platform_dispatcher.dart (hunk 1 above)
Comment (ditman) [reply lib/web_ui/lib/src/engine/platform_dispatcher.dart:401]: Android seems to ignore `null` primary color. [Source](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/ActivityManager.java#1257). It falls-back to [default `ActivityTaskDescription_colorPrimary`](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/Activity.java#5100), which I guess is defined in some App theme (being in R)

**Comment by gaaclarke** [comment]
@ditman This PR appears to possibly have broken some customer tests.  See https://github.com/flutter/flutter/pull/84395#issuecomment-859173990  Can you take a look?  The failure may be a flake, if it shows up again again in the next roll we should revert this PR.

**Comment by gaaclarke** [comment]
After further investigation this PR is off the hook.  There was a change in a different repository that appears to have caused the issue.

**Comment by ditman** [comment]
Shoot, I've just seen this notification. Thanks for clearing this PR @gaaclarke!

**Pull request merged**